### PR TITLE
Add experimental Mage-Flow and Mage-Flow-Edit support

### PR DESCRIPTION
**Pull request by sdbds**
## Summary

- add experimental Mage-Flow T2I and Mage-Flow-Edit LoRA training, caching, and image generation
- port the packed NR-MMDiT, MageVAE, and Qwen3-VL conditioning used by Microsoft Mage commit `ea7109b`
- load the released Comfy-Org single-file DiT, Qwen3-VL, and MageVAE weights with strict header validation
- load tokenizer, chat-template, and image-processor assets automatically from a pinned `microsoft/Mage-Flow/text_encoder` revision; no processor/tokenizer CLI is required
- keep the existing bucket/collator path while preserving the cumulative-length packed ABI for future heterogeneous native-resolution packing
- support one to three ordered Edit references, SDPA and optional FlashAttention 2
- reuse Musubi facilities for scaled FP8, gradient checkpointing, block swap, and `torch.compile`

## Compatibility

- architecture identities and 16-pixel bucket steps are additive and per-architecture
- public dataset collation and existing architecture behavior are unchanged
- dependency pins in `pyproject.toml` are unchanged
- loaders validate safetensors headers before constructing modules and instantiate the 4.1B DiT skeleton on `meta`
- unsupported `--compile_fullgraph` is rejected before model loading; regular `--compile` remains supported with graph breaks

## Validation

- `python -m pytest -q`: 172 passed, 1 skipped (optional FlashAttention 2 gate)
- `ruff check`: passed for all 35 Python files changed from upstream main
- `ruff format --check`: all 35 changed Python files formatted
- `compileall` and `git diff --check`: passed
- released Comfy-Org checkpoint headers: DiT 397 BF16 tensors, MageVAE 839 BF16 tensors, and Qwen3-VL 713 BF16 tensors, matching the implemented strict contracts
- real `AutoProcessor.from_pretrained` smoke at the pinned official revision produced `Qwen3VLProcessor`, `Qwen2TokenizerFast`, and `Qwen2VLImageProcessorFast`

## Current limitations

- support remains experimental until full released-weight numerical parity and GPU end-to-end validation are recorded
- the first release uses existing resolution buckets; it does not add a native token-budget batch scheduler
- each large component currently requires one regular `.safetensors` file
- INT8 ConvRot checkpoints are not supported
- LoRA scope is fixed to the supported repeated transformer-block linear layers

Official reference: https://microsoft.github.io/Mage/flow/  
Official model: https://huggingface.co/microsoft/Mage-Flow  
Single-file weights: https://huggingface.co/Comfy-Org/Mage-Flow  
Pinned source: https://github.com/microsoft/Mage/tree/ea7109b3515ddd995c2e1212656dc1bc3a9607b7

### CODE REVIEW

**Comment by sdbds** [comment]
tested result:
image

reference:
```
--num_cpu_threads_per_process=8
  --mixed_precision=bf16
  --downcast_bf16 musubi-tuner\src\musubi_tuner\mage_flow_train_network.py
  --dit=./ckpts/diffusion_models/mage_flow_bf16.safetensors
  --vae=./ckpts/vae/mage_flow_vae_bf16.safetensors
  --text_encoder=./ckpts/text_encoder/qwen_3_VL_4b.safetensors
  --output_dir=./output_dir
  --network_module=musubi_tuner.networks.lora_mage_flow
  --learning_rate=8e-5
  --lr_scheduler=cosine_with_min_lr
  --mixed_precision=bf16
  --max_train_epochs=20
  --gradient_accumulation_steps=1
  --timestep_sampling=uniform_shift
  --discrete_flow_shift=6.0
  --weighting_scheme=none
  --lr_decay_steps=0.2
  --lr_scheduler_num_cycles=1
  --lr_scheduler_min_lr_ratio=0.1
  --network_dim=64
  --network_alpha=32
  --max_grad_norm=1.0
  --max_data_loader_n_workers=8
  --seed=1026
  --output_name=mage_flow_lora_qinglong
  --cuda_cudnn_benchmark
  --cuda_allow_tf32
  --persistent_data_loader_workers
  --gradient_checkpointing
  --logging_dir=./logs
  --log_with=tensorboard
  --save_every_n_epochs=2
  --sample_at_first
  --sample_every_n_epochs=2
  --sample_prompts=./toml/qinglong_mage_flow.txt
  --flash_attn
  --optimizer_type=adv_optm.AdamW_adv
  --optimizer_args betas=.95,.98
```

sample:
`--w 1024 --h 1024 --d 1026 --l 5 --s 28 --f 1 --fs 6`

I think we're ready to merge.
~The only issue is that the generated images look a bit rough; I'm not sure if this is caused by the MageFlow VAE.~
fixed

@kohya-ss
